velox/vector/BaseVector.cpp

```diff
@@ -673,6 +673,12 @@ bool isLazyNotLoaded(const BaseVector& vector) {
     case VectorEncoding::Simple::CONSTANT:
       return vector.valueVector() ? isLazyNotLoaded(*vector.valueVector())
                                   : false;
+    case VectorEncoding::Simple::ROW: {
```

Comment:
Should we apply similar logic to arrays and maps?

Reply:
Sure, we can. I'm not sure if we actually create lazies for them. Let me investigate the fuzzer first

Reply:
AFAIK, arrays and maps never use lazyVectors as value vectors when generated from a scan (@Yuhta please correct me if I am wrong). Even if they get generated during expression eval, the column would have been loaded before being rolled into a complex vector.

Reply:
Got it. Thank you for checking.

Reply:
Array and map elements are never generated lazily.  It's only struct fields without nulls can be read lazily.